source/renderer/app/components/staking/stake-pools/hooks/useInViewPort.tsx

```diff
@@ -0,0 +1,27 @@
+import { useRef, useState, useCallback } from 'react';
+
+export const useInViewPort = () => {
+  const [isInViewport, setIsInViewport] = useState(true);
+  const targetRef = useRef(null);
+  const observerRef = useRef(
+    new IntersectionObserver((entries) => {
```

Comment:
Just to check, do we need also need to `unobserve` when we remove the component from the DOM?